core/settings/qPrefUnit.cpp

```diff
@@ -104,9 +104,35 @@ void qPrefUnits::set_unit_system(unit_system_values value)
 	if (value == METRIC) {
 		prefs.unit_system = METRIC;
 		prefs.units = SI_units;
+
+		// the different types of measurements needs to be adjusted as well.
+		// set the type to the opposite to guarantee, the signal get sent
+		prefs.units.volume = units::VOLUME::CUFT;
+		set_volume(units::VOLUME::LITER);
+		prefs.units.weight = units::WEIGHT::LBS;
+		set_weight(units::WEIGHT::KG);
+		prefs.units.length = units::LENGTH::FEET;
+		set_length(units::LENGTH::METERS);
+		prefs.units.pressure = units::PRESSURE::PSI;
+		set_pressure(units::PRESSURE::BAR);
+		prefs.units.temperature = units::TEMPERATURE::FAHRENHEIT;
+		set_temperature(units::TEMPERATURE::CELSIUS);
```

Comment:
that code looks really, really weird.
I'm not sure I have a better idea to ensure that the signals get sent... maybe the `set_xxx()` functions should have an optional argument to force sending the signal? But this just looks crazy.

Also, the commit message refers to `unit_meterChanged()` which should be `unit_lengthChanged`, I hope.

Reply:
I have added a bigger comment explaining why it is needed, and also added a longer text in the commit.

It was my judgement that changing the macros (with possible side effects) or writing all these setters directly was a far bigger change.

Yes, my errorr, changed to unit_lengthChanged